lib/entry/LiveEntry.js

```diff
@@ -190,5 +183,18 @@ LiveEntry.prototype.toJSON = function () {
     return diag;
 };
 
+LiveEntry.prototype.cleanup = function() {
+
+    var flavorsPromises = _.map(this.flavorsDownloaders, (flavor) => {
+        return flavor.stop();
+    });
+
+    return Q.all(flavorsPromises)
+        .then(() => {
+            this.logger.info("Stopping PlaylistGenerator");
+            return this.playlistGenerator.stop();
+        });
+}
+
```

Comment:
As part of state machine redesign PLAT-5431 your suggestion can be implemented. In our discussion with Guy today, it was suggested to add Abort state that can be set from any state and is irreversible and moves the state machine directly to final state, stop.